src/foundry/client/pixi/webgl/shaders/samplers/base-sampler.d.mts

```diff
@@ -0,0 +1,215 @@
+import type { ConstructorOf } from "../../../../../../types/utils.d.mts";
+
+export {};
+
+declare global {
+  /**
+   * A simple shader to emulate a PIXI.Sprite with a PIXI.Mesh (but faster!)
+   */
+  class BaseSamplerShader extends AbstractBaseShader {
+    static override create<T extends BaseSamplerShader>(
+      this: ConstructorOf<T>,
+      defaultUniforms?: AbstractBaseShader.Uniforms | undefined,
+    ): T;
+
+    /**
+     * The plugin name associated for this instance, if any.
+     * Returns "batch" if the shader is disabled.
+     */
+    get pluginName(): (typeof BaseSamplerShader)["classPluginName"];
+
+    /**
+     * The named batch sampler plugin that is used by this shader, or null if no batching is used.
+     * @defaultValue `"batch"`
+     */
+    static classPluginName: string | null;
```

Comment:
Refactor this type out into the namespace for nicer reusability? It's kind of ugly in hints to see the whole `ReturnType` stuff specified.